### PR TITLE
Adds Survey Functionality to WFJT

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
This addresses #5901. It also adds tests

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

##### AWX VERSION
```
9.3
```

### CODE REVIEW

File: awx/ui_next/src/api/models/WorkflowJobTemplates.js [hunk 1]
```diff
@@ -49,7 +49,21 @@ class WorkflowJobTemplates extends SchedulesMixin(NotificationsMixin(Base)) {
...
+    return this.http.get(`${this.baseUrl}${id}/access_list/`, {
+      params,
+    });
+  }
+
+  readSurvey(id) {
+    return this.http.get(`${this.baseUrl}${id}/survey_spec/`);
+  }
+
+  updateSurvey(id, survey = null) {
```
Comment (jlmitch5) [inline awx/ui_next/src/api/models/WorkflowJobTemplates.js]: would this not be survey = {}?

**Review by jlmitch5 — APPROVED** [review]
looks good other than potentially wrong initial vale for the survey spec post (not sure)

File: awx/ui_next/src/api/models/WorkflowJobTemplates.js (hunk 1 above)
Comment (keithjgrant) [reply awx/ui_next/src/api/models/WorkflowJobTemplates.js]: I'm not sure we need any default value here... seems like it's an essential parameter

Comment (AlexSCorey) [reply awx/ui_next/src/api/models/WorkflowJobTemplates.js]: we don't need that default value at all.  Good catch.  I'll remove it from here and from `JobTemplates.js`

**Pull request merged**